tests/integration/src/test/scala/mesosphere/marathon/integration/MesosAppIntegrationTest.scala

```diff
@@ -12,20 +12,39 @@ import mesosphere.marathon.raml.{App, Container, DockerContainer, EngineType}
 import mesosphere.marathon.state.PathId._
 import mesosphere.marathon.state.{HostVolume, VolumeMount}
 import mesosphere.mesos.Constraints.hostnameField
-import mesosphere.{AkkaIntegrationTest, WhenEnvSet, WaitTestSupport}
+import mesosphere.{AkkaIntegrationTest, WaitTestSupport, WhenEnvSet}
+import org.scalatest.concurrent.PatienceConfiguration.Interval
 import play.api.libs.json.JsObject
 
 import scala.collection.immutable.Seq
 import scala.concurrent.duration._
 
 class MesosAppIntegrationTest extends AkkaIntegrationTest with EmbeddedMarathonTest {
 
+  override lazy val mesosNumMasters = 1
+  override lazy val mesosNumSlaves = 2
+
+
+  override def afterAll(): Unit = {
+    // We need to start all the agents for the teardown to be able to kill all the (UNREACHABLE) executors/tasks
+    mesosCluster.agents.foreach(_.start())
+    eventually { mesosCluster.state.value.agents.size shouldBe mesosCluster.agents.size }
+    super.afterAll()
+  }
+
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    mesosCluster.agents(1).stop()
```

Comment:
should we also make sure that agent(0) is started? this seems dangerous, because we are not sure that at least one agent is always running...

Reply:
`agent(0)` is started in MesosClusterTest